src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -1,353 +0,0 @@
-                        route = function.Name;
-                    }
-
-                    WebJobsRouteBuilder builder = function.Metadata.IsProxy ? proxiesRoutesBuilder : routesBuilder;
-                    builder.MapFunctionRoute(function.Metadata.Name, route, constraints, function.Metadata.Name);
-                }
-            }
-
-            IRouter proxyRouter = null;
-            IRouter functionRouter = null;
-
-            if (proxiesRoutesBuilder.Count > 0)
-            {
-                 proxyRouter = proxiesRoutesBuilder.Build();
-            }
-
-            if (routesBuilder.Count > 0)
-            {
-                functionRouter = routesBuilder.Build();
-            }
-
-            _router.AddFunctionRoutes(functionRouter, proxyRouter);
-        }
-
-        public override void Shutdown()
-        {
-            string message = "Environment shutdown has been triggered. Stopping host and signaling shutdown.";
-            Instance?.Logger.LogInformation(message);
-
-            Stop();
-
-            Program.InitiateShutdown();
-        }
-
-        public Task DelayUntilHostReady()
-        {
-            // ensure that the host is ready to process requests
-            return DelayUntilHostReady(_hostTimeoutSeconds, _hostRunningPollIntervalMilliseconds);
-        }
-
-        public Task<bool> DelayUntilHostReady(int timeoutSeconds = ScriptConstants.HostTimeoutSeconds, int pollingIntervalMilliseconds = ScriptConstants.HostPollingIntervalMilliseconds, bool throwOnFailure = true)
-        {
-            return DelayUntilHostReady(this, timeoutSeconds, pollingIntervalMilliseconds, throwOnFailure);
-        }
-
-        internal static async Task<bool> DelayUntilHostReady(ScriptHostManager hostManager, int timeoutSeconds = ScriptConstants.HostTimeoutSeconds, int pollingIntervalMilliseconds = ScriptConstants.HostPollingIntervalMilliseconds, bool throwOnFailure = true)
-        {
-            await Utility.DelayAsync(timeoutSeconds, pollingIntervalMilliseconds, () =>
-            {
-                return !hostManager.CanInvoke() &&
-                        hostManager.State != ScriptHostState.Error;
-            });
-
-            bool hostReady = hostManager.CanInvoke();
-
-            if (throwOnFailure && !hostReady)
-            {
-                throw new HttpException(HttpStatusCode.ServiceUnavailable, "Function host is not running.");
-            }
-
-            return hostReady;
-        }
-
-        internal static async Task<bool> DelayUntilHostReady(WebHostResolver resolver)
-        {
-            // need to ensure the host startup has been initiated
-            var manager = resolver.GetWebScriptHostManager();
-            var tIgnore = manager.EnsureHostStarted(CancellationToken.None);
-
-            // wait for the host to get into a running state
-            return await manager.DelayUntilHostReady(throwOnFailure: false);
-        }
-
-        internal static async Task DelayUntilEnvironmentReady()
```

Comment:
Where did this move to? This logic was put in place to address Linux specialization issues.

Reply:
This is still in place and used by the middleware. This became an extension method as it works with any `IScriptHostManager` implementation.